doc/source/tutorial/stats/sampling_tdr.rst

```diff
@@ -0,0 +1,190 @@
+
+.. _sampling-tdr:
+
+Transformed Density Rejection (TDR)
+===================================
+
+.. currentmodule:: scipy.stats
+
+* Required: T-concave PDF, dPDF
+* Optional: mode, center
+* Speed:
+
+  * Set-up: slow
+  * Sampling: fast 
+
+
+TDR is an acceptance/rejection method that uses the concavity of a transformed
+density to construct hat function and squeezes automatically. Such PDFs are
+called T-concave. Currently the following transformations are implemented:
+
+.. math::
+
+    c = 0.: T(x) &= \log(x)\\
+    c = -0.5: T(x) &= \frac{1}{\sqrt{x}} \text{ (Default)}
+
+In addition to the PDF, it also requires the derivative of the PDF w.r.t x
+(i.e. the variate). These functions must be present as methods of a python
+object which can then be passed to the generators to instantiate their object.
+
+Three variants of this method are available:
+
+* GW: squeezes between construction points.
+* PS: squeezes proportional to hat function. (Default)
+* IA: same as variant PS but uses a composition method with
+  "immediate acceptance" in the region below the squeeze.
+
+This can be changed by passing a ``variant`` parameter.
+
+An example of using this method is shown below:
+
+    >>> from scipy.stats import TransformedDensityRejection
+    >>> from scipy.stats import norm
+    >>> 
+    >>> class StandardNormal:
+    ...     def pdf(self, x):
+    ...         # note that the normalization constant is not required
+    ...         return np.exp(-0.5 * x*x)
+    ...     def dpdf(self, x):
+    ...         return -x * np.exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> 
+    >>> urng = np.random.default_rng(0xbb51a838c8a087854ad3643b2b268d43)
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+    >>> rng.rvs()
+    0.7771420061983989
+
+In the above example, we have used the TDR method to sample from the standard
+normal distribution. Note that we can drop the normalization constant while
+computing the PDF. This usually helps speed up the sampling stage. Also, note
+that the PDF doesn't need to be vectorized. It should accept and return a
+scalar.
+
+It is also possible to evaluate the inverse of the CDF of the hat distribution
+directly using the ``ppf_hat`` method.
+
+    >>> rng.ppf_hat(0.5)
+    -0.00018050266342362759
+    >>> norm.ppf(0.5)
+    0.0
+    >>> u = np.linspace(0, 1, num=10)
+    >>> rng.ppf_hat(u)
+    array([       -inf, -1.22227372, -0.7656556 , -0.43135731, -0.14002921,
+            0.13966423,  0.43096141,  0.76517113,  1.22185606,         inf])
+    >>> norm.ppf(u)
+    array([       -inf, -1.22064035, -0.76470967, -0.4307273 , -0.1397103 ,
+            0.1397103 ,  0.4307273 ,  0.76470967,  1.22064035,         inf])
+
+Apart from the PPF method, other attributes can be accessed
+to see how well the generator fits the given distribution. These are:
+
+* 'sqhratio': (area below squeeze) / (area below hat) for the generator. It
```

Comment:
Is it too much to name this `squeeze_hat_ratio`? Would be more explicit IMO.

Reply:
I personally prefer more explicit names. But if it's just me, no problem. Nothing blocking here.

Reply:
I think I agree with @tupui: since we try to use more descriptive names, it might be better to write it out. but I am also fine to keep it as it is

Reply:
Thanks for your feedback both. I have changes `sqhratio` to `squeeze_hat_ratio` now.

Reply:
Resolving this as it has been addressed. Feel free to unresolve if you want to discuss this further!